addons/base_action_rule/models/base_action_rule.py

```diff
@@ -0,0 +1,338 @@
+        required=True, domain=[('osv_memory', '=', False)])
+    model = fields.Char(related='model_id.model')
+    create_date = fields.Datetime(string='Create Date', readonly=True)
+    active = fields.Boolean(
+        default=True,
+        help="When unchecked, the rule is hidden and will not be executed.")
+    sequence = fields.Integer(help="Gives the sequence order when displaying a list of rules.")
+    kind = fields.Selection([
+        ('on_create', 'On Creation'),
+        ('on_write', 'On Update'),
+        ('on_create_or_write', 'On Creation & Update'),
+        ('on_time', 'Based on Timed Condition')],
+        string='When to Run')
+    trg_date_id = fields.Many2one(
+        'ir.model.fields', string='Trigger Date',
+        help="When should the condition be triggered. If present, will be checked by the scheduler. If empty, will be checked at creation and update.",
+        domain="[('model_id', '=', model_id), ('ttype', 'in', ('date', 'datetime'))]")
+    trg_date_range = fields.Integer(
+        'Delay after trigger date',
+        help="Delay after the trigger date."
+        "You can put a negative number if you need as delay before the"
+        "trigger date, like sending a reminder 15 minutes before a meeting.")
+    trg_date_range_type = fields.Selection([
+        ('minutes', 'Minutes'), ('hour', 'Hours'),
+        ('day', 'Days'), ('month', 'Months')],
+        'Delay type', default='day')
+    trg_date_calendar_id = fields.Many2one(
+        'resource.calendar', string='Use Calendar',
+        help='When calculating a day-based timed condition, it is possible to use a calendar to compute the date based on working days.',
+        ondelete='set null')
+    act_user_id = fields.Many2one('res.users', string='Set Responsible')
+    act_followers = fields.Many2many("res.partner", string="Add Followers")
+    server_action_ids = fields.Many2many(
+        'ir.actions.server', string='Server Actions',
+        domain="[('model_id', '=', model_id)]",
+        help="Examples: email reminders, call object service, etc.")
+    filter_pre_id = fields.Many2one(
+        'ir.filters', string='Before Update Filter',
+        ondelete='restrict', domain="[('model_id', '=', model_id.model)]",
+        help="If present, this condition must be satisfied before the update of the record.")
+    filter_pre_domain = fields.Char(string='Before Update Domain', help="If present, this condition must be satisfied before the update of the record.")
+    filter_id = fields.Many2one(
+        'ir.filters', string='Filter',
+        ondelete='restrict', domain="[('model_id', '=', model_id.model)]",
+        help="If present, this condition must be satisfied before executing the action rule.")
+    filter_domain = fields.Char(string='Domain', help="If present, this condition must be satisfied before executing the action rule.")
+    last_run = fields.Datetime(string='Last Run', readonly=True, copy=False)
```

Comment:
You can remove the string parameter here, as the ORM will generate a capitalized version of the field name by default